src/context/AuthUserContext .tsx

```diff
@@ -0,0 +1,33 @@
+import { createContext, useContext } from 'react';
+import { User } from "firebase/auth";
+import { useUserAuth } from "@/hooks/useAuth.ts";
+import { UserType } from "@/types/User.ts";
+
+const AuthContext = createContext<{
+    authUserInfo: UserType | User | null,
+    authUserIsLoading: boolean
+}>({
+    authUserInfo: null,
+    authUserIsLoading: true
+})
+
+export const AuthContextProvider = ({ children }: { children: React.ReactNode }) => {
+    const { authUserInfo, authUserIsLoading }: { authUserInfo: UserType | User | null, authUserIsLoading: boolean } = useUserAuth()
+
+    return (
+        <AuthContext.Provider
+            value={{ authUserInfo, authUserIsLoading }}
+        >
+            {children}
+        </AuthContext.Provider>
+    )
+}
+
+// To be used in child components of AuthContextProvider that needs the AuthContext
+export const useAuthContext = () => {
```

Comment:
Tous les hooks doivent se retrouver dans le même dossier.
Donc il serait préférable que tu déplaces la logique de ton hook customisé useAuthContext à l'intérieur du dossier src/hooks. Et bien sur de ne pas oublier de l'importer dans les fichiers correspondants :p

Je l'ai surtout remarqué car il y a une erreur/warning qui s'affiche sur mon IDE. Oui je suis allée faire un tour sur ta branche, perso je trouve ça plus simple pour détecter les petits soucis que simplement checker les fichiers modifiés direct sur Github.

Il s'agit de ce message "Fast refresh only works when a file only exports components. Use a new file to share constants or functions between components.eslint(react-refresh/only-export-components)"

Reply:
Oui tu as raison merci Caroline :)